### PR TITLE
fix: Print args when iOS tests are not found

### LINKED ISSUE #1388: Output inconsistency (ios vs android)
**Describe the bug**
> A clear and concise description of what the bug is.

There are differences in flank output for iOS and Android when all test are filtered
Below is the output for https://github.com/Flank/flank/blob/master/integration_tests/src/test/kotlin/integration/AllTestFilteredIT.kt
* iOS https://gist.github.com/pawelpasterz/e517272563a0bcbf83fb4fdbcf94d549
* android https://gist.github.com/pawelpasterz/5a4d8498629b716abd86cd0b70ee4b2e

**To Reproduce**
> Steps to reproduce the behavior:

1. run ios test with config 
    ```
    gcloud:
      test: [path to zip]
      xctestrun-file: [path to xctestrun]
    flank:
      test-targets:
        - nonExisting/Class
    ```
1. run android test with config 
    ```
    gcloud:
      app: [path to app apk]
      test: [path to test apk]
      test-targets:
        - class non.existing.Class
    flank:
      disable-sharding: true
    ```

**Expected behavior**
> A clear and concise description of what you expected to happen.

There is a decision to be made if flank should (or not) print args.
Based on the answer:
* fix should be implemented
* IT must be updated

**Details (please complete the following information):**
> Have you tested on the latest Flank snapshot?
>
> Post the output of ` flank --version`.

```
version: local_snapshot
revision: 2d2982474e9d38ee39639ed7282b84fb5e33ce01
```

**Additional context**
> Add any other context about the problem here.

Introduced in https://github.com/Flank/flank/pull/1219

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #1388 

1. run ios test with config

```
gcloud:
  test: [path to zip]
  xctestrun-file: [path to xctestrun]
flank:
  test-targets:
    - nonExisting/Class
```

1. Flank should print args before any validation (ios and android)
1. Exit code should be 1
1. Integration tests should works

## Checklist

- [X] Integration tests works

### CODE REVIEW

**Review by jan-gogo — CHANGES_REQUESTED** [review]
I don't like the validation is split between execution steps. Wouldn't be better to print args before any validation instead of breaking validation consistency?

**Comment by adamfilipow92** [comment]
> I don't like the validation is split between execution steps. Wouldn't be better to print args before any validation instead of breaking validation consistency?
> 
> WDYT? @pawelpasterz @piotradamczyk5 @adamfilipow92

Agree

**Comment by pawelpasterz** [comment]
> I don't like the validation is split between execution steps. Wouldn't be better to print args before any validation instead of breaking validation consistency?
> 
> WDYT? @pawelpasterz @piotradamczyk5 @adamfilipow92

+1

I think printing args (always) is a good idea.

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidRunCommand.kt [hunk 1]
```diff
@@ -47,8 +48,7 @@ class AndroidRunCommand : CommonRunCommand(), Runnable {
             MockServer.start()
         }
 
-        AndroidArgs.load(Paths.get(configPath), cli = this).validate().run {
-            setupLogLevel()
+        AndroidArgs.load(Paths.get(configPath), cli = this).logArgs().validate().run {
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidRunCommand.kt:51]: 👍

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosRunCommand.kt [hunk 2]
```diff
@@ -60,3 +60,8 @@ class IosRunCommand : CommonRunCommand(), Runnable {
     )
     var dumpShards: Boolean = false
 }
+
+private fun IosArgs.logArgs() = also {
+    setupLogLevel()
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosRunCommand.kt]: `setupLogLevel` looks like an independent operation not only related to args printing. IMO it should be called before `IosArgs.load`.

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidRunCommand.kt [hunk 3]
```diff
@@ -62,3 +62,8 @@ class AndroidRunCommand : CommonRunCommand(), Runnable {
     )
     var dumpShards: Boolean = false
 }
+
+private fun AndroidArgs.logArgs() = also {
+    setupLogLevel()
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidRunCommand.kt]: Same here

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosRunCommand.kt (hunk 2 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/cli/firebase/test/ios/IosRunCommand.kt]: Changed! Thanks!

File: test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidRunCommand.kt (hunk 3 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/cli/firebase/test/android/AndroidRunCommand.kt]: Changed! Thanks!

**Review by pawelpasterz — APPROVED** [review]

**Review by Sloox — APPROVED** [review]
It appears it will fix https://github.com/Flank/flank/issues/1389

**Review by cryptoyang — APPROVED** [review]

**Review by jan-gogo — APPROVED** [review]

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**